File: front-end/src/app/shared/models/summoner-model.ts
import MatchModel from './match-model';

export default class SummonerModel {
    accountId: string;
    id: number;
    name: string;
    profileIconId: number;
    puuid: string;
    revisionDate: number;
    summonerLevel: number;
    matches: MatchModel[];
}

File: back-end/LeagueAppAPI/LeagueAppAPI/Models/SummonerModel.cs
using System.Collections.Generic;

namespace LeagueAppAPI.Models
{
    public class SummonerModel
    {
        public string AccountId { get; set; }
        public string Id { get; set; }
        public string Name { get; set; }
        public string ProfileIconId { get; set; }
        public string Puuid { get; set; }
        public string RevisionDate { get; set; }
        public string SummonerLevel { get; set; }
        public List<MatchModel> Matches { get; set; }
    }
}
